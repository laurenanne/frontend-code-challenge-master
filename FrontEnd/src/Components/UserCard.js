import React from "react";
import {
  Card,
  CardBody,
  CardTitle,
  CardText,
  ListGroup,
  ListGroupItem,
  Button,
  CardHeader,
} from "reactstrap";

function UserCard({ users }) {
  return (
    <ListGroup className="user-list col-md-8 d-inline-block">
      {users.map((u) => (
        <ListGroupItem
          style={{ border: "none" }}
          className="m-2 text-center"
          key={u.id}
        >
          <Card style={{ borderColor: "grey" }} color="white">
            <CardHeader>Id: {u.id}</CardHeader>
            <CardBody color="secondary">
              <CardTitle>
                {u.firstName} {u.lastName}
              </CardTitle>
              <CardText>{u.email}</CardText>
              <Button style={{ backgroundColor: "#8f5cd1" }}>{u.state}</Button>
            </CardBody>
          </Card>
        </ListGroupItem>
      ))}
    </ListGroup>
  );
}

export default UserCard;
