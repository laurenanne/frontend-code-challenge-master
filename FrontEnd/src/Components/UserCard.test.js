import React from "react";
import { render } from "@testing-library/react";
import UserCard from "./UserCard";
import { MemoryRouter } from "react-router";

it("matches snapshot", function() {
  let users = [
    {
      id: "1",
      firstName: "First",
      lastName: "Last",
      email: "myemail@email.com",
      state: "pending",
    },
  ];
  const { asFragment } = render(
    <MemoryRouter>
      <UserCard users={users} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
