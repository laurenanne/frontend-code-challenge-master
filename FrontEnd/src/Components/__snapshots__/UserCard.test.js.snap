// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="user-list col-md-8 d-inline-block list-group"
  >
    <li
      class="m-2 text-center list-group-item"
    >
      <div
        class="card bg-white"
        style="border-color: grey;"
      >
        <div
          class="card-header"
        >
          Id: 1
        </div>
        <div
          class="card-body"
          color="secondary"
        >
          <div
            class="card-title"
          >
            First Last
          </div>
          <p
            class="card-text"
          >
            myemail@email.com
          </p>
          <button
            class="btn btn-secondary"
            style="background-color: rgb(143, 92, 209);"
          >
            pending
          </button>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
